crates/sui-graphql-rpc/src/mutation.rs

```diff
@@ -71,27 +78,72 @@ impl Mutation {
             );
         }
         let transaction = Transaction::from_generic_sig_data(tx_data, sigs);
+        let options = SuiTransactionBlockResponseOptions::new()
+            .with_events()
+            .with_balance_changes()
+            .with_raw_input()
+            .with_raw_effects();
 
         let result = sui_sdk_client
             .quorum_driver_api()
             .execute_transaction_block(
                 transaction,
-                SuiTransactionBlockResponseOptions::default(),
-                Some(ExecuteTransactionRequestType::WaitForEffectsCert),
+                options,
+                // This needs to be WaitForLocalExecution because we need the transaction effects.
+                // TODO: make it possible to execute without waiting for local execution?
+                Some(ExecuteTransactionRequestType::WaitForLocalExecution),
             )
             .await
             // TODO: use proper error type as this could be a client error or internal error
             // depending on the specific error returned
             .map_err(|e| Error::Internal(format!("Unable to execute transaction: {e}")))
             .extend()?;
 
+        let raw_effects: NativeTransactionEffects = bcs::from_bytes(&result.raw_effects)
+            .map_err(|e| Error::Internal(format!("Unable to deserialize transaction effects: {e}")))
+            .extend()?;
+        let sender_signed_data: SenderSignedData = bcs::from_bytes(&result.raw_transaction)
+            .map_err(|e| Error::Internal(format!("Unable to deserialize transaction data: {e}")))
+            .extend()?;
+
+        let events = result
+            .events
+            .ok_or(Error::Internal(
+                "No events are returned from tranasction execution".to_string(),
+            ))?
```

Comment:
Isn't it possible for a transaction to not emit events though?

Reply:
Even when a transaction does not emit events, since we have `show_events` turned on it should still return a `Some(SuiTransactionBlockEvents(Vec<SuiEvent>))` with an empty vector inside, hence the `ok_or` here that would convert None to an error. I understand the confusion though and will add some comments.